server/src/main/java/org/apache/cassandra/sidecar/acl/authentication/JwtParameterExtractor.java

```diff
@@ -0,0 +1,142 @@
+ */
+public class JwtParameterExtractor
+{
+    protected static final String SITE_SUFFIX = "/.well-known/openid-configuration";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JwtParameterExtractor.class);
+    private static final String SITE_PARAM_KEY = "site";
+    private static final String CLIENT_ID_PARAM_KEY = "client_id";
+    private static final String SCOPE_SEPARATOR_PARAM_KEY = "scope_separator";
+    private static final String DEFAULT_SCOPE_SEPARATOR = ",";
+    private static final String SCOPES_SUPPORTED_PARAM_KEY = "scopes_supported";
+    private static final String CONFIG_DISCOVER_INTERVAL_PARAM_KEY = "config_discover_interval";
+    private static final SecondBoundConfiguration DEFAULT_CONFIG_DISCOVER_INTERVAL
+    = SecondBoundConfiguration.parse("1m");
+
+    private final String site;
+    private final String clientId;
+    private final SecondBoundConfiguration configDiscoverInterval;
+    private final List<String> scopes;
+
+    public JwtParameterExtractor(Map<String, String> parameters)
+    {
+        validate(parameters);
+        this.site = removeSiteSuffix(parameters);
+        this.clientId = parameters.get(CLIENT_ID_PARAM_KEY);
+        this.scopes = buildScopes(parameters);
+        this.configDiscoverInterval = parameters.containsKey(CONFIG_DISCOVER_INTERVAL_PARAM_KEY)
+                                      ? SecondBoundConfiguration.parse(parameters.get(CONFIG_DISCOVER_INTERVAL_PARAM_KEY))
+                                      : DEFAULT_CONFIG_DISCOVER_INTERVAL;
+    }
+
+    /**
+     * @return site to dynamically retrieve the configuration information of an OpenID provider
+     */
+    public String site()
+    {
+        return site;
+    }
+
+    /**
+     * @return clientId is a unique identifier used to identity applications/users
+     */
+    public String clientId()
+    {
+        return clientId;
+    }
+
+    /**
+     * @return scopes granted to a user
+     */
+    public List<String> scopes()
+    {
+        return scopes;
+    }
+
+    /**
+     * @return interval at which {@link io.vertx.ext.auth.oauth2.providers.OpenIDConnectAuth} discover is called to
+     * dynamically retrieve configuration information of an OpenID provider.
+     */
+    public SecondBoundConfiguration configDiscoverInterval()
+    {
+        return configDiscoverInterval;
+    }
+
+    private void validate(Map<String, String> parameters)
+    {
+        if (parameters == null)
+        {
+            throw new IllegalArgumentException("JWT parameters can not be null");
+        }
+
+        validateParameterPresence(parameters, SITE_PARAM_KEY);
+        validateParameterPresence(parameters, CLIENT_ID_PARAM_KEY);
+    }
+
+    private void validateParameterPresence(Map<String, String> parameters, String paramKey)
+    {
+        if (!parameters.containsKey(paramKey) || isNullOrEmpty(parameters.get(paramKey)))
```

Comment:
minor NIT, no need to check if it doesn't contain. doing a get should be sufficient
```suggestion
        if (isNullOrEmpty(parameters.get(paramKey)))
```